projects/plugins/jetpack/extensions/extended-blocks/paid-blocks/with-upgrade-banner.jsx

```diff
@@ -21,6 +23,15 @@ export default createHigherOrderComponent(
 			return <BlockEdit { ...props } />;
 		}
 
+		/**
+		 * The banner is displayed in the jetpack/payment-buttons.
+		 *
+		 * TODO: Move this validation outside of this section since this is a generic component.
+		 */
+		if ( name === 'jetpack/recurring-payments' ) {
+			return <BlockEdit { ...props } />;
+		}
```

Comment:
This prevents individual Payment Button block from having the banner.

While they are not insertable from the block inserter, some patterns (such as the Payments Block pattern) inserts it:

Pattern | Before | After
--- | --- | ---
<img width="349" alt="Screenshot 2022-12-20 at 10 47 38" src="https://user-images.githubusercontent.com/1233880/208636873-d7200960-119b-40ef-b6af-4cdc9f18c632.png"> | <img width="571" alt="Screenshot 2022-12-20 at 10 47 57" src="https://user-images.githubusercontent.com/1233880/208636985-1a3b5366-898f-4994-a977-1f8126f6fb51.png">| <img width="595" alt="Screenshot 2022-12-20 at 10 50 49" src="https://user-images.githubusercontent.com/1233880/208637015-18105628-33ec-4dd5-8b15-34977883ea45.png">

Reply:
Oh, good point! I assumed it's always used inside payment buttons. Thanks! :)